### PR TITLE
Adds reusable loading and error view controllers

### PR DESCRIPTION
**Pull request by rhoque-glx**
**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #

**Related issues that should be LINKED to this PR:**
* Connects #

### CODE REVIEW

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            
+            if let retryButtonString = retryButtonString {
+                retryButton.setTitle(retryButtonString, for: .normal)
+            }
+        }
+        
+        if let errorTitleString = errorTitleString {
+            errorTitle.text = errorTitleString
+        }
+        else {
```
Comment (benRmc) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Wrong line for this else statement.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            errorTitle.isHidden = true
+            errorMessageToErrorTitleVerticalConstraint.constant = 0
+        }
+        
+        if let errorMessageString = errorMessageString {
```
Comment (benRmc) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Don't need this `if let` unwrapping. `errorMessage.text` itself is optional.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    var errorTitleString: String?
+    var errorMessageString: String?
+    var retryButtonString: String?
+    
+    public override func viewDidLoad() {
+        if delegate == nil {
+            retryButton.isHidden = true
+            retryButton.isEnabled = false
+            
+            if let retryButtonString = retryButtonString {
```
Comment (benRmc) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Don't need this `if let`.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/LoadingViewController.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import UIKit
+
+open class LoadingViewController: UIViewController {
+    @IBOutlet weak var loadingTitle: UILabel!
+    
+    var loadingTitleString: String?
+    
+    public override func viewDidLoad() {
+        if let loadingTitleString = loadingTitleString {
```
Comment (benRmc) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/LoadingViewController.swift]: Don't need this `if let`.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        if let errorMessageString = errorMessageString {
+            errorMessage.text = errorMessageString
+        }
+    }
+
+    @IBAction func didRetry(_ sender: UIButton) {
+        delegate?.errorViewControllerDidRetry(self)
+    }
+    
+    public func set(title: String) {
```
Comment (benRmc) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Is setTitle(_ title: String) better? @OmarPadierna @ata-n

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift (hunk 2 above)
Comment (rhoque-glx) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Don't want to override default storyboard text unless errorMessageString is set

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift (hunk 3 above)
Comment (rhoque-glx) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Don't want to override default storyboard text unless retryButtonString is set

File: traveler-ios-ui/TravelerKitUI/ViewControllers/LoadingViewController.swift (hunk 4 above)
Comment (rhoque-glx) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/LoadingViewController.swift]: Don't want to override default storyboard text unless loadingTitleString is set

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift (hunk 1 above)
Comment (rhoque-glx) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: fixed

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift (hunk 3 above)
Comment (OmarPadierna) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Doesn't matter anyway, the if-let is inside the if that checks for the delegate to nil, which means that if the delegate does not exist you will hide the retry button. The consequence being that you're changing the title of a hidden button.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift (hunk 5 above)
Comment (OmarPadierna) [reply traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: this function is unnecessary. You can just pass the string you want to set in errorTitleString in the prepare for segue method when instantiating this view controller. We don't have setters in any of our view controllers and therefore this breaks our established convention. One of our goals is to have the codebase look like it was written by a single person.

File: traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        errorTitleString = title
+    }
+    
+    public func set(message: String) {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift]: Same as comment above.

**Pull request merged**